class.wp-cldr.php

```diff
@@ -483,4 +472,161 @@ public function first_day_of_week( $country ) {
 		}
 		return '';
 	}
+
+	/**
+	 * Gets the currency used in each country worldwide.
+	 *
+	 * @link http://www.iso.org/iso/country_codes ISO 3166 country codes
+	 * @link http://www.iso.org/iso/currency_codes ISO 4217 currency codes
+	 *
+	 * @return array An associative array of ISO 3166 country codes and the ISO 4217 code for the currency currently used in each country.
+	 */
+	public function get_currency_for_all_countries() {
+		$json_file = $this->get_locale_bucket( 'supplemental', 'currencyData' );
+
+		// This CLDR item has a history of all the currencies ever used in a country
+		// so we need to loop through them to find one without a `_to` ending date
+		// and without a `_tender` flag which are always false indicating
+		// the currency wasn't legal tender.
+		foreach ( $json_file['supplemental']['currencyData']['region'] as $territory => $currencies ) {
+			foreach ( $currencies as $currency_dates ) {
+				if ( ! array_key_exists( '_to', current( $currency_dates ) ) && ! array_key_exists( '_tender', current( $currency_dates ) ) ) {
+					$result[ $territory ] = key( $currency_dates );
+				}
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Gets the currency currently used in a country.
+	 *
+	 * @link http://www.iso.org/iso/country_codes ISO 3166 country codes
+	 * @link http://www.iso.org/iso/currency_codes ISO 4217 currency codes
+	 *
+	 * @param string $country A two-letter ISO 3166-1 country code.
+	 * @return string The three-letter ISO 4217 code for the currency currently used in that country.
+	 */
+	public function get_currency_for_country( $country ) {
```

Comment:
yeah it feels like we're a bit inconsistent with caching.  Let's discuss in #62.